### PR TITLE
validator-network: Fix multiple DHT queries for the same validator ID

### PR DESCRIPTION
**Pull request by jsdanielh**
- Fix the `get_validator_peer_id` function to properly handle the case where no cache entry is present once it obtains the write lock. This avoids multiple DHT queries for the same validator ID when there are multiple requests being made and no cache entry is present.
- Only clear the `PeerId` cache on errors if the faced error is for the same `PeerId` that is store in the cache. This avoids unnecessary clears for already properly resolved `PeerId`s.

## Pull request checklist

- [X] All tests pass. The project builds and runs.
- [X] I have resolved any merge conflicts.
- [X] I have resolved all `clippy` and `rustfmt` warnings.

### CODE REVIEW

File: validator-network/src/network_impl.rs [hunk 1]
```diff
@@ -195,9 +195,10 @@ where
                 if let Some(validator_key) =
                     self.validator_keys.read().get(usize::from(validator_id))
                 {
-                    self.validator_peer_id_cache
-                        .write()
-                        .remove(validator_key.compressed());
+                    self.validator_peer_id_cache.write().insert(
+                        validator_key.compressed().clone(),
+                        CacheState::Error(NetworkError::UnknownValidator(validator_id)),
+                    );
```
Comment (hrxi) [inline validator-network/src/network_impl.rs]: This should probably only be done if this invalidates the cache entry that led to the error. Would that be possible?

Comment (jsdanielh) [reply validator-network/src/network_impl.rs]: Not sure if I'm following you, this is trying to invalidate the cache entry that led to the error

Comment (hrxi) [reply validator-network/src/network_impl.rs]: I'm imagining the following scenario:

The current cache entry of validator 1 is invalid. Two requests to that are currently in flight. The first one fails and clears the cache entry. Another request is sent, causing the cache to be updated to the correct value. The second initial request fails, and now clears the cache value even though it's already been updated.

Comment (jsdanielh) [reply validator-network/src/network_impl.rs]: Yeah, thanks! Updated PR to incorporate the change we discussed

**Review by hrxi — APPROVED** [review]
Looks good to me. :)

**Pull request merged**